client/test/client.go

```diff
@@ -112,3 +116,28 @@ func (c *Client) SignTx(tx *tee.Transaction) {
 		log.Panicf("Error signing tx: %v", err)
 	}
 }
+
+// call calls the given function, waits for the TX to be mined and updates the
+// clients last mined blocknumber.
+func (c *Client) call(ctx context.Context, call func(context.Context, *bind.TransactOpts) (*types.Transaction, error)) error {
```

Comment:
Now that we have the context in `tr`, you can remove it from the `call func` parameters :see_no_evil:

Reply:
Oh, right.